osu.Game/OsuGame.cs

```diff
@@ -1189,6 +1194,14 @@ protected override void LoadComplete()
                 if (mode.NewValue != OverlayActivation.All) CloseAllOverlays();
             };
 
+            backButtonVisibility.ValueChanged += visible =>
+            {
+                if (visible.NewValue)
+                    BackButton.Show();
+                else
+                    BackButton.Hide();
+            };
```

Comment:
Doesn't this sort of setup mean it's possible to put yourself in the situation where the back button is in view but the screen actually isn't "user-exitable" or however you'd put it?

Reply:
That is true, but for the sake of simplicity, it should be `OsuScreen`'s responsibility to never show the back button if it explicitly declares to not be exitable. Otherwise, I will have to account for `AllowUserExit` in this bindable and make this bindable always be triggered when a screen is changed, and that just increases the complexity and takes us back to step one which is "why is this being done differently".

At the moment, I've made `OsuScreen` decide the initial state of the back button appearance based on whether it is declared to be user-exitable or not:
https://github.com/ppy/osu/blob/3e1b4f4ac564a1b69b2d8111b19d3908f99980e4/osu.Game/Screens/OsuScreen.cs#L59-L62